plugins/modules/aci_fabric_interface_policy_group.py

```diff
@@ -0,0 +1,359 @@
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    policy_group_type = module.params.get("type")
+    dwdm_policy = module.params.get("dwdm_policy")
+    link_level_policy = module.params.get("link_level_policy")
+    link_flap_policy = module.params.get("link_flap_policy")
+    l3_interface_policy = module.params.get("l3_interface_policy")
+    macsec_policy = module.params.get("macsec_policy")
+    monitoring_policy = module.params.get("monitoring_policy")
+    transceiver_policy_tdn = module.params.get("transceiver_policy_tdn")
+
+    if policy_group_type == "leaf":
+        policy_group_class_name = "fabricLePortPGrp"
+        policy_group_class_rn = "leportgrp-{0}".format(name)
+    else:
+        policy_group_class_name = "fabricSpPortPGrp"
+        policy_group_class_rn = "spportgrp-{0}".format(name)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fabric",
+            aci_rn="fabric",
+        ),
+        subclass_1=dict(
+            aci_class="fabricFuncP",
+            aci_rn="funcprof",
+        ),
+        subclass_2=dict(
+            aci_class=policy_group_class_name,
+            aci_rn=policy_group_class_rn,
+            module_object=name,
+            target_filter={"name": name},
+        ),
+        child_classes=[
+            "fabricRsDwdmFabIfPol",
+            "fabricRsFIfPol",
+            "fabricRsFLinkFlapPol",
+            "fabricRsL3IfPol",
+            "fabricRsMacsecFabIfPol",
+            "fabricRsMonIfFabricPol",
+            "fabricRsOpticsFabIfPol",
+        ],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if dwdm_policy is not None:
+            child_configs.append(dict(fabricRsDwdmFabIfPol=dict(attributes=dict(tnDwdmFabIfPolName=dwdm_policy))))
+        if link_level_policy is not None:
+            child_configs.append(dict(fabricRsFIfPol=dict(attributes=dict(tnFabricFIfPolName=link_level_policy))))
+        if link_flap_policy is not None:
+            child_configs.append(dict(fabricRsFLinkFlapPol=dict(attributes=dict(tnFabricFLinkFlapPolName=link_flap_policy))))
+        if l3_interface_policy is not None:
+            child_configs.append(dict(fabricRsL3IfPol=dict(attributes=dict(tnL3IfPolName=l3_interface_policy))))
+        if macsec_policy is not None:
+            child_configs.append(dict(fabricRsMacsecFabIfPol=dict(attributes=dict(tnMacsecFabIfPolName=macsec_policy))))
+        if monitoring_policy is not None:
+            child_configs.append(dict(fabricRsMonIfFabricPol=dict(attributes=dict(tnMonFabricPolName=monitoring_policy))))
+        if transceiver_policy_tdn is not None:
+            child_configs.append(dict(fabricRsOpticsFabIfPol=dict(attributes=dict(tDn=transceiver_policy_tdn))))
```

Comment:
fabricRsOpticsFabIfPol not supported in all versions

Reply:
Changed the logic to support lower versions